server/index.js

```diff
@@ -25,23 +25,65 @@ passport.deserializeUser(function(user, done) {
 })
 // });
 
+/* Miles' code..making some adjustments below in order to add new users to the DB */
+// passport.use(new Strategy({
+//     clientID: config.FACEBOOK_clientID,
+//     clientSecret: config.FACEBOOK_clientSecret,
+//     callbackURL: 'http://localhost:3000/auth/facebook/callback'
+//   },
+//   function(accessToken, refreshToken, profile, done) {
+//     // User's Facebook profile is supplied as the user record
+
+//     User.findOrCreate({ facebookId: profile.id }, function (err, user) {
+//       return cb(err, user);
+//     });
+//     console.log(...arguments);
+//     done(null, profile);
+//   }
+// ));
+
 passport.use(new Strategy({
     clientID: config.FACEBOOK_clientID,
     clientSecret: config.FACEBOOK_clientSecret,
     callbackURL: 'http://localhost:3000/auth/facebook/callback'
   },
   function(accessToken, refreshToken, profile, done) {
     // User's Facebook profile is supplied as the user record
-
-    // User.findOrCreate({ facebookId: profile.id }, function (err, user) {
-    //   return cb(err, user);
-    // });
-    console.log(...arguments);
-    done(null, profile);
-  }
-));
-
-
+    process.nextTick(function() {
+
+    User.findOne({ 'facebookId' : profile.id }, function(err, user) {
+
+    // if there is an error, stop everything and return that
+    // ie an error connecting to the database
+    if (err) {
+      return done(err);
+    }
+
+    if (user) {
+      return done(null, user); // user found, return that user
+    } else {
+      // if there is no user found with that facebook id, create them
+        var newUser            = new User();
+        // set all of the facebook information in our user model
+        newUser.facebookId  = profile.id; // set the users facebook id                   
+        newUser.favRecipes = {}; //we set the user's favorite recipes to a blank object
+        newUser.weeks = [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}]; // I thought we'd start with a blank object for each day. Then, if a user adds a recipe to a meal for a day we'll begin adding new key-value pairs here
```

Comment:
@jmaxc6 How are you planning to change the recipes from week to week? Just delete the final seven indices, scoot the first seven indices over, and create seven new objects?

An alternative to that is grouping the recipes into two arrays - one for each week. At the start of a new week, we could say that array2 = array1.slice(), then delete the values of every object key in array 1.